web/api/v1/api.go

```diff
@@ -464,6 +475,66 @@ func (api *API) queryRange(r *http.Request) (result apiFuncResult) {
 	}, nil, res.Warnings, qry.Close}
 }
 
+func (api *API) queryExemplars(r *http.Request) apiFuncResult {
+	ctx := r.Context()
+	if to := r.FormValue("timeout"); to != "" {
+		var cancel context.CancelFunc
+		timeout, err := parseDuration(to)
+		if err != nil {
+			err = errors.Wrap(err, "invalid parameter 'timeout'")
+			return apiFuncResult{nil, &apiError{errorBadData, err}, nil, nil}
+		}
+
+		ctx, cancel = context.WithTimeout(ctx, timeout)
+		defer cancel()
+	}
+
+	q, err := api.Queryable.Querier(ctx, timestamp.FromTime(minTime), timestamp.FromTime(maxTime))
+	if err != nil {
+		return apiFuncResult{nil, &apiError{errorBadData, err}, nil, nil}
+	}
+
+	expr, err := parser.ParseExpr(r.FormValue("query"))
+	if err != nil {
+		return apiFuncResult{nil, &apiError{errorBadData, err}, nil, nil}
+	}
+
+	selectors, err := parser.ExtractSelectors(expr)
+	if err != nil {
+		return apiFuncResult{nil, &apiError{errorBadData, err}, nil, nil}
+	}
+	if len(selectors) < 1 {
+		return apiFuncResult{nil, &apiError{errorBadData, fmt.Errorf("no series found for supplied query: %s", expr)}, nil, nil}
+	}
+
+	var retExemplars []exemplarData
+
+	for _, selectorArr := range selectors {
+		ss := q.Select(false, nil, selectorArr...)
+
+		eq, err := api.ExemplarQueryable.Querier(context.Background())
+		if err != nil {
+			return apiFuncResult{nil, &apiError{errorBadData, err}, nil, nil}
+		}
+		for {
+			if ok := ss.Next(); !ok {
+				break
+			}
+
+			res, err := eq.Select(ss.At().Labels())
```

Comment:
Don't assume at is not using resources, I would get at := ss.At()

Reply:
`Don't assume at is not using resources` not sure what you mean